### ISSUE TITLE
`run` output doesn't get spliced into transcripts

### ISSUE THREAD

**Issue by atacratic** [unisonweb/unison#1172]
The output of `run` commands goes to the wrong place during transcript parsing.

Input transcript: 
```
    ```unison
    hi = '(printLine "Hello world!")
    ```

    ```ucm
    .> add
    .> run hi
    ```
```

Console output during transcript parsing:

```

  Transcript will be run on a new, empty codebase.

  ⚠️
  
  Initializing a new codebase in: /home/chris/unison/transcript-53622049952cfb9b.

   _____     _             
  |  |  |___|_|___ ___ ___ 
  |  |  |   | |_ -| . |   |
  |_____|_|_|_|___|___|_|_|
  
  Running the provided transcript file...
  

⚙️   Processing stanza 2 of 3.Hello world!
⚙️   Processing stanza 3 of 3.
💾  Wrote /home/chris/unison/transcript-run.output.md
```

Note this wrongly includes the "Hello world!" output.

Transcript output produced:

```
    ```unison
    hi = '(printLine "Hello world!")
    ```

    ```ucm

      I found and typechecked these definitions in scratch.u. If you
      do an `add` or `update`, here's how your codebase would
      change:
      
        ⍟ These new definitions are ok to `add`:
        
          hi : ∀ (). () ->{IO} ()
       
      Now evaluating any watch expressions (lines starting with
      `>`)... Ctrl+C cancels.

    ```
    ```ucm
    .> add

      ⍟ I've added these definitions:
      
        hi : ∀ (). () ->{IO} ()

    .> run hi

    ```
```
This output wrongly omits the "Hello world!"

**Comment by aryairani** [comment]
The program literally prints to `stdout`, it will be pretty hairy to splice it into the transcript.

**Comment by anovstrup** [comment]
@aryairani At least in principle, couldn't the transcript runner use its own IO handler (one that handles `printLine` requests differently)? Not sure how much refactoring this would entail, but that might have other benefits anyway (e.g., paving the way to supporting IO testing)...

**Comment by aryairani** [comment]
@anovstrup Yeah maybe something like that could work.  `IO` is currently baked into the runtime, but we could conceivably factor it differently.

**Comment by aryairani** [comment]
We're in a better position to do this today than we were at the time, and I think it would be great.

I guess it's still a little tricky because we're not just launching a process that we can tap into its IO; we're ... something something the Runtime in our same current thread? I'm not sure.
